src/qrp/mod.rs

```diff
@@ -6,6 +6,31 @@ use uuid::Uuid;
 pub mod cerved_qrp;
 pub mod qrp_service;
 
+#[derive(Clone)]
+pub enum QrpGeneration {
+    ByTaxCode(String),
+    #[deprecated]
+    ByVatNumber(String),
+}
+
+impl From<QrpGeneration> for String {
+    fn from(query: QrpGeneration) -> Self {
+        match query {
+            QrpGeneration::ByTaxCode(tax_code) => tax_code,
+            QrpGeneration::ByVatNumber(vat_number) => vat_number,
+        }
+    }
+}
+
+impl Display for QrpGeneration {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            QrpGeneration::ByTaxCode(tax_code) => write!(f, "tax_code: {}", tax_code),
+            QrpGeneration::ByVatNumber(vat_number) => write!(f, "vat_number: {}", vat_number),
```

Comment:
```suggestion
            QrpGeneration::ByVatNumber(vat_number) => write!(f, "vat_code: {}", vat_number),
```
We usually use this naming ☝️